isisd/isis_te.c

```diff
@@ -422,12 +434,55 @@ static int isis_mpls_te_adj_ip_disabled(struct isis_adjacency *adj, int family,
 		return 0;
 	}
 
+	return 0;
+}
+
+static int isis_mpls_te_adj_ip_disabled(struct isis_adjacency *adj, int family,
+					bool global)
+{
+	int ret;
+
+	/* Sanity Check */
+	if (!adj || !adj->circuit || !adj->circuit->ext)
+		return 0;
+
+	ret = _isis_mpls_te_adj_ip_disabled(adj, family, global);
+
 	/* Update LSP */
-	lsp_regenerate_schedule(circuit->area, circuit->is_type, 0);
+	lsp_regenerate_schedule(adj->circuit->area, adj->circuit->is_type, 0);
 
-	return 0;
+	return ret;
 }
 
+static void isis_mpls_te_circuit_ip_update(struct isis_circuit *circuit)
+{
+	struct isis_adjacency *adj;
+
+	if (circuit->circ_type != CIRCUIT_T_P2P)
+		return;
+
+	adj = circuit->u.p2p.neighbor;
+
+	if (!adj)
+		return;
+
+	if (adj->ipv4_address_count > 0)
+		_isis_mpls_te_adj_ip_enabled(adj, AF_INET, false);
+	else
+		_isis_mpls_te_adj_ip_disabled(adj, AF_INET, false);
+
+	if (adj->ll_ipv6_count > 0)
+		_isis_mpls_te_adj_ip_enabled(adj, AF_INET6, false);
+	else
+		_isis_mpls_te_adj_ip_disabled(adj, AF_INET6, false);
+
```

Comment:
You could remove this: 1/ as mention in the RFC, link local MUST NOT be used for IP address (both local and remote) except for unnumbered interface which is not supported by current implementation and 2/ in _isis_mpls_te_adj_ip_enabled() line 370-371 and in _isis_mpls_te_adj_ip_disabled() line 430 a test is perform against global . Thus a call to these function with link local will do nothing except consume some CPU cycle ;-)

Reply:
1/ => OK removed.
2/ Cannot be removed.